### PR TITLE
Raise phpstan level to `max`

### PR DESCRIPTION
**Pull request by yhabteab**

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
+
+        - '#Call to an undefined method Icinga\\Data\\SimpleQuery::.*#'
+
+        - '#Call to an undefined method Icinga\\Data\\ConnectionInterface::getResource\(\)#'
+
+        - '#Call to an undefined method Icinga\\Module\\Cube\\Ido\\IdoCube::addDimensionByName\(\)#'
+
+        - '#Call to an undefined method Icinga\\Module\\Cube\\Cube::getSortBy\(\)#'
+        - '#Access to undefined constant Icinga\\Module\\Cube\\Cube::DIMENSION_SEVERITY_SORT_PARAM#'
+
```
Comment (sukhwinder33445) [inline phpstan.neon:35]: Please fix or add them to baseline.

**Review by sukhwinder33445 — CHANGES_REQUESTED** [review]

File: phpstan.neon [hunk 2]
```diff
@@ -1,16 +1,35 @@
...
+                - '#. but return statement is missing#'
+            reportUnmatched: false
+
+        - '#Call to an undefined method Icinga\\Module\\Cube\\Ido\\ZfSelectWrapper::joinLeft\(\)#'
+
+        - '#Call to an undefined method Icinga\\Data\\SimpleQuery::.*#'
+
+        - '#Call to an undefined method Icinga\\Data\\ConnectionInterface::getResource\(\)#'
+
+        - '#Call to an undefined method Icinga\\Module\\Cube\\Ido\\IdoCube::addDimensionByName\(\)#'
```
Comment (sukhwinder33445) [inline phpstan.neon:31]: I think these can also be included in the baseline. If the bug pops up later, we want to fix it.

**Review by sukhwinder33445 — CHANGES_REQUESTED** [review]

File: phpstan.neon (hunk 2 above)
Comment (yhabteab) [reply phpstan.neon:31]: Why? These are just some interface method & magic method call errors that don't need to be fixed.

**Review by sukhwinder33445 — APPROVED** [review]

**Review by nilmerg — APPROVED** [review]

File: .github/workflows/php.yml [hunk 3]
```diff
@@ -31,20 +31,21 @@ jobs:
...
-          && git clone --single-branch --branch snapshot/nightly https://github.com/Icinga/icinga-php-thirdparty.git vendor/icinga-php-thirdparty
+        run: |
+          composer require -n --no-progress overtrue/phplint
+          git clone --depth 1 https://github.com/Icinga/icingaweb2.git vendor/icingaweb2
+          git clone --depth 1 https://github.com/Icinga/icingadb-web.git vendor/modules/icingadb-web
+          git clone --depth 1 -b snapshot/nightly https://github.com/Icinga/icinga-php-library.git vendor/icinga-php-library
+          git clone --depth 1 -b snapshot/nightly https://github.com/Icinga/icinga-php-thirdparty.git vendor/icinga-php-thirdparty
 
       - name: PHP Lint
         if: success() || matrix.allow_failure
```
Comment (nilmerg) [inline .github/workflows/php.yml:42]: change this as well pls

**Review by nilmerg — APPROVED** [review]

**Pull request merged**